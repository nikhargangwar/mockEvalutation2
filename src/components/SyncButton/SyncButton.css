.Mainbody{
    height: 80vh;
    display: flex;
    justify-content: center;
    align-items: center;
}
.sync-button-wrapper{
    width: 600px;
    /* border: 1px solid black; */
    height: 200px;
    display: flex;
    justify-content: space-around;
    align-items: center;
    flex-direction: column;
    margin: 20px;
    font-weight: 100;
}


.sync-text{
    font-size: 30px;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.sync-button{
    height: 40px;
    width: 150px;
    border-radius: 100px;
    background-color: #4C63AA ;
    color:white;
    margin-bottom: 20px;
    border: none;
}