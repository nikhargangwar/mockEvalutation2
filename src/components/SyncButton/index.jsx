import './SyncButton.css';
import React from 'react';
import { useNavigate } from 'react-router-dom';
import { SONGS } from '../../constants/routesPaths';

function SyncButton() {
  const navigate = useNavigate();
  const syncButtonHandler = () => {
    console.log('jo');
    navigate(SONGS);
  };

  return (
    <div className="Mainbody">
      <div className="sync-button-wrapper">
        <div className="sync-text">
          :((
          <br />
          seems a bit empty in here...
        </div>
        <div>
          <button data-testid="sync-button" className="sync-button" type="button" onClick={syncButtonHandler}><h2>sync</h2></button>
        </div>
      </div>
    </div>
  );
}

export default SyncButton;
