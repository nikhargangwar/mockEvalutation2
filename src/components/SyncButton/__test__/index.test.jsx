import React from 'react';
// import { useNavigate } from 'react-router-dom';
import {
  render, screen, fireEvent, waitFor,
} from '@testing-library/react';
import SyncButton from '..';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

describe('SyncButton', () => {
  it('should render when mounted', () => {
    const { container } = render(<SyncButton />);
    expect(container).toMatchSnapshot();
  });

  it('should have sync button when mounted', () => {
    render(<SyncButton />);
    const syncButton = screen.getByTestId('sync-button');
    expect(syncButton).toBeTruthy();
  });

  it('should navigate when sync buttonn is clicked', async () => {
    render(<SyncButton />);
    expect(mockNavigate).not.toBeCalled();
    const syncButton = screen.getByTestId('sync-button');
    fireEvent.click(syncButton);
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
});
