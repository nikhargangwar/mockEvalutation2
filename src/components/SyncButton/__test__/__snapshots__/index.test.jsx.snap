// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncButton should render when mounted 1`] = `
<div>
  <div
    class="Mainbody"
  >
    <div
      class="sync-button-wrapper"
    >
      <div
        class="sync-text"
      >
        :((
        <br />
        seems a bit empty in here...
      </div>
      <div>
        <button
          class="sync-button"
          data-testid="sync-button"
          type="button"
        >
          <h2>
            sync
          </h2>
        </button>
      </div>
    </div>
  </div>
</div>
`;
